go/cmd/dolt/commands/sqlserver/sqlserver.go

```
@@ -432,3 +436,51 @@ func setupDoltConfig(dEnv *env.DoltEnv, apr *argparser.ArgParseResults, config s
 
 	return nil
 }
+
+func generateYamlConfigIfNone(
+	ap *argparser.ArgParser,
+	help cli.UsagePrinter,
+	args []string,
+	dEnv *env.DoltEnv,
+	serverConfig servercfg.ServerConfig) error {
+	const yamlConfigName = "config.yaml"
+
+	specifiesConfigFile, err := argsSpecifyServerConfigFile(ap, help, args)
+	if err != nil {
+		return err
+	}
+
+	if specifiesConfigFile {
+		return nil
+	}
+
+	path := filepath.Join(serverConfig.DataDir(), yamlConfigName)
+	exists, isDir := dEnv.FS.Exists(path)
+	if exists {
+		if isDir {
+			cli.PrintErrf("Couldn't generate YAML config at %s: directory with same name already exists", path)
```

This should just be silently ignored